### PR TITLE
Dropdown: selectedKey can now be null

### LINKED ISSUE #8093: Controlled Dropdown doesn't update state when selectedKey is undefined
### Please provide a reproduction of the bug in a codepen:
 https://codepen.io/isabelacmor/pen/bzXdEw
 
#### Actual behavior:
Setting Dropdown's selectedKey to undefined doesn't update the items selected in `selectedIndices`

#### Expected behavior:
`selectedIndices` is updated when selectedKey is undefined.

#### Priorities and help requested:
 
Are you willing to submit a PR to fix? Yes
 
Requested priority: High
 
Products/sites affected: All Dropdowns

#### Root cause:
setState is ignored when selectedKey is undefined.
(thanks to @brandonthomas for finding the root cause!)

```
if (newProps[selectedKeyProp] !== undefined && (newProps[selectedKeyProp] !== this.props[selectedKeyProp] || didOptionsChange)) {
this.setState({
selectedIndices: this._getSelectedIndexes(newProps.options, newProps[selectedKeyProp])
});
}
```
 
https://github.com/OfficeDev/office-ui-fabric-react/blob/867c9f9237ca6d802a7fb2e5401868dbff434988/packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx#L131

### PR DESCRIPTION
**Pull request by dzearing**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #8093
- [X] Include a change request file using `$ npm run change`

#### Description of changes

You can now clear the dropdown on demand.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8099)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.test.tsx [hunk 1]
```diff
@@ -118,6 +118,19 @@ describe('Dropdown', () => {
       expect(titleElement.textContent).toEqual('2');
     });
 
+    it('does clear when the selectedKey is null', () => {
```
Comment (cliffkoh) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.test.tsx]: Shouldn't combo box have the same test given that it also takes in selectedKey?

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.types.ts [hunk 2]
```diff
@@ -86,7 +86,7 @@ export interface IDropdownProps extends ISelectableDroppableTextProps<IDropdown,
    * Keys of the selected items. If you provide this, you must maintain selection
    * state by observing onChange events and passing a new value in when changed.
    */
```
Comment (cliffkoh) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.types.ts]: This should also say ` Note that passing in `null` will cause selection to be reset.`

File: common/changes/office-ui-fabric-react/dropdown-types_2019-02-23-01-54.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "office-ui-fabric-react",
+      "comment": "Dropdown: `selectedKey` can now take null to clear the selection.",
```
Comment (cliffkoh) [inline common/changes/office-ui-fabric-react/dropdown-types_2019-02-23-01-54.json]: ComboBox is also affected by this change right?

Comment (ecraig12345) [reply common/changes/office-ui-fabric-react/dropdown-types_2019-02-23-01-54.json]: ComboBox doesn't inherit from Dropdown. Though ideally it ought to follow the same behavior.

**Comment by aneeshack4** [comment]
Gentle ping on this - looks like there are a few minor changes before this can be merged.

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.test.tsx (hunk 1 above)
Comment (dzearing) [reply packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.test.tsx]: ok done

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.types.ts (hunk 2 above)
Comment (dzearing) [reply packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.types.ts]: fixed

**Comment by dzearing** [comment]
@cliffkoh This ended up not being auto-merged, any ideas?

**Pull request merged**